src/librustc_save_analysis/lib.rs

```diff
@@ -228,6 +232,7 @@ impl<'l, 'tcx: 'l> SaveContext<'l, 'tcx> {
                     visibility: From::from(&item.vis),
                     docs: docs_for_attrs(&item.attrs),
                     sig: self.sig_base(item),
+                    attributes: remove_docs_from_attrs(&item.attrs),
```

Comment:
Certainly. I followed the usage of `docs_for_attrs` here, which does something very similar, except for comments instead of attributes. They are even more similar now, because attributes are lowered to (String, Span) pairs. Calling this function (or even the lowering) earlier reduces the amount of data which needs to be copies around.